### PR TITLE
Add TDE files to extract headline data into queryable tables

### PR DESCRIPTION
**Pull request by jacksonj04**
This forms the basis of bulk data extraction for analysis; two new Template Driven Extraction templates will populate `documents.summary` and `documents.propertysummary` tables, which can be collectively queried using an SQL `JOIN` to extract a table of key information about documents.

## Fields

These are a first pass, and will likely change as we get a better handle on the stats we need to extract.

Fields which have `<invalid-values>` set to `ignore` and `<nullable>` set to `true` will create rows with `NULL` values where a value cannot be found. Rows where this isn't the case will be dropped, as they will upset the `JOIN`.

### Documents

| Field | Value | Ignore invalid? |
| --- | --- | --- |
| `uri` | `xdmp:node-uri(.)` | No |
| `ncn` | `sem:iri(//uk:cite//text())` | Yes |
| `name` | `sem:iri(//akn:FRBRWork/akn:FRBRname/@value)` | Yes |

### Properties

| Field | Value | Ignore invalid? |
| --- | --- | --- |
| `uri` | `xdmp:node-uri(.)` | No |
| `doc_uri` | `dls:version/dls:document-uri` | Yes |
| `version_number` | `dls:version/dls:version-id` | Yes |
| `modified` | `prop:last-modified` | Yes |
| `published` | `published` | Yes |

### CODE REVIEW

File: src/main/ml-schemas/sql-document-extract.xsd [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  <rows>
+    <row>
+      <schema-name>documents</schema-name>
+      <view-name>summary</view-name>
+      <columns>
+        <column>
+          <name>uri</name>
+          <scalar-type>string</scalar-type>
+          <val>xdmp:node-uri(.)</val>
+          <reindexing>visible</reindexing>
```
Comment (dragon-dxw) [inline src/main/ml-schemas/sql-document-extract.xsd:30]: no ignoring of invalid value, but I suppose an invalid value is probably impossible?

File: src/main/ml-schemas/sql-document-extract.xsd [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          <val>sem:iri(//uk:cite//text())</val>
+          <nullable>true</nullable>
+          <invalid-values>ignore</invalid-values>
+          <reindexing>visible</reindexing>
+        </column>
+        <column>
+          <name>name</name>
+          <scalar-type>string</scalar-type>
+          <val>sem:iri(//akn:FRBRWork/akn:FRBRname/@value)</val>
+          <reindexing>visible</reindexing>
```
Comment (dragon-dxw) [inline src/main/ml-schemas/sql-document-extract.xsd:44]: here an invalid value might be more possible -- or something without a name. Would be interesting to know the reasons for the inconsistency.

**Review by dragon-dxw — APPROVED** [review]

File: src/main/ml-schemas/sql-document-extract.xsd (hunk 1 above)
Comment (jacksonj04) [reply src/main/ml-schemas/sql-document-extract.xsd:30]: Yes; a document with no URI makes no sense, so we want these to fail.

File: src/main/ml-schemas/sql-document-extract.xsd (hunk 2 above)
Comment (jacksonj04) [reply src/main/ml-schemas/sql-document-extract.xsd:44]: Good catch 👍🏻

**Pull request merged**